internal/dataplane/deckgen/generate.go

```diff
@@ -48,7 +48,7 @@ func ToDeckContent(
 
 		for _, r := range s.Routes {
 			route := file.FRoute{Route: r.Route}
-			fillRoute(&route.Route)
+			fillRoute(&route.Route, k8sState.ExpressionRoutes)
```

Comment:
I think that instead of putting the `ExpressionRoutes` feature flag into `KongState`, we could create some struct (e.g. `DeckContentGeneratingParams`) that would include it, as well as `schemas`, `selectorTags`, and `formatVersion` that are all passed to `ToDeckContent`. This way we'd keep the separation of the configuration that was generated by the Parser from the parameters that steer how the deck file is generated out of this config. WDYT?

Reply:
done